blockchain_connector/avalon_blockchain_connector/ethereum/ethereum_connector.py

```diff
@@ -120,27 +123,58 @@ def _retrieve_first_worker_details(self):
             sys.exit(1)
         return worker_id, worker_info["result"]
 
-    def _add_worker_to_chain(self, worker_id, worker_info):
+    def _add_update_worker_to_chain(self, worker_id_hex, worker_info):
         """
-        This function adds a worker to the Ethereum blockchain
+        This function adds/updates a worker in the Ethereum blockchain
         """
-        worker_id = self._eth_client.get_bytes_from_hex(worker_id)
+        worker_id = self._eth_client.get_bytes_from_hex(worker_id_hex)
         worker_type = worker_info["workerType"]
         org_id = self._eth_client\
             .get_bytes_from_hex(worker_info["organizationId"])
         app_type_id = self._eth_client.get_bytes_from_hex(
             worker_info["applicationTypeId"])
         details = json.dumps(worker_info["details"])
-
-        txn_dict = self._worker_reg_contract_instance.functions.workerRegister(
-            worker_id, worker_type, org_id, [app_type_id], details)\
-            .buildTransaction(self._eth_client.get_transaction_params())
+        workers = self._lookup_workers_on_chain()
+        if worker_id_hex in workers:
+            logging.info("Updating worker {} on ethereum blockchain"\
+                .format(worker_id_hex))
+            txn_dict = self._worker_reg_contract_instance.functions\
+                .workerUpdate(worker_id, details)\
+                .buildTransaction(self._eth_client.get_transaction_params())
+        else:
+            logging.info("Adding new worker {} to ethereum blockchain"\
+                .format(worker_id_hex))
+            txn_dict = self._worker_reg_contract_instance.functions\
+                .workerRegister(worker_id, worker_type, org_id,
+                                [app_type_id], details)\
+                .buildTransaction(self._eth_client.get_transaction_params())
```

Comment:
Can we move this buildTransaction to workerRegister()?

Reply:
@Ram-srini, I will accommodate this in another PR. Replacing these with sdk calls would be the way to go.